include/lsst/afw/math/statistics/Validators.h

```diff
@@ -0,0 +1,87 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_MATH_STATISTICS_VALIDATORS_H
+#define LSST_AFW_MATH_STATISTICS_VALIDATORS_H
+
+#include "lsst/afw/image/MaskedImage.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+namespace statistics {
+
```

Comment:
It appears that `Validator.operator()` should take 4 arguments: value, mask, variance, and weight. Please document that clearly somewhere in this file. (Initially I thought the variadic arguments were related to `PropertySet`!). Also, I suggest listing them explicitly; clarity at the expense of verbosity.  Thus, for example, I think `ValidateRange` would be clearer as:
```
    template <typename Value, typename Mask, typename Variance, typename Weight>
    bool operator()(Value val, Mask, Variance, Weight) const {
        Value tmp = fabs(val - _center);
        return (tmp <= _limit);
    }
```
It's tricky having 4 arguments and getting the order right. Listing them ought to help (though of course there is no way to prevent listing them in the wrong order).

Reply:
Why are validators listed as internal (the `@internal` tag in the doc string)? They appear to be part of the public API. For example `SinglePassStatistics` accepts a user-specified validator, and I assume it is public.